core/src/main/java/io/grpc/ForwardingChannelBuilder.java

```diff
@@ -242,6 +244,18 @@ public T proxyDetector(ProxyDetector proxyDetector) {
     return thisT();
   }
 
+  @Override
+  public T defaultServiceConfig(@Nullable Map<String, ?> serviceConfig) {
```

Comment:
The  problem of `Map<String, Object>` might be: If the user has a `Map<String, Map>` object at hand, they have to cast it to `Map<String, Object>` first and then pass it to the API. Actually I'm not too strong about it. @carl-mastrangelo you want to debate about it?

Reply:
I don't disagree, but given the service config options, what is the likelihood that they set values that are the same type? It seems they would create the object just for passing to this call, so it would also be trivial for them to use Object instead of Map<String, Object> as the value.

I could accept having it, it just seems like it would be more confusing to users without much benefit.

Reply:
I found that `Map<String, ?>` involved less internal casting, and was easier to pass in.  It also means that it isn't possible to write to the map since no type matches the wildcard, implying it is immutable.

Reply:
How could `Map<String, ?>` involve less internal casting? Can you give an example?

Reply:
Here is a similar case with `List<Object>`:  https://github.com/grpc/grpc-java/blob/v1.19.x/core/src/main/java/io/grpc/internal/ServiceConfigUtil.java#L501

It's not always going to happen that way, but still it was nice to avoid that.  

One other thing, when passing in this type, it's nice to pass values into Map<String, ?>, especially in unit tests where the type comes from an inlined value:

```java
defaultServiceConfig(ImmutableMap.of("methodConfig", myTestMethodConfig));
```

It involves a cast to the `Map<String, Object>`.  I think we should try it as `Map<String, ?>` and consider if it should be changed in API review.